flocker/node/testtools.py

```diff
@@ -226,5 +226,5 @@ def to_node(node_state):
     :return Node: Equivalent node.
     """
     return Node(uuid=node_state.uuid, hostname=node_state.hostname,
-                applications=node_state.applications,
-                manifestations=node_state.manifestations)
+                applications=node_state.applications or [],
+                manifestations=node_state.manifestations or {})
```

Comment:
Mmm this is interesting.  Maybe this change isn't required now that the various test data has been updated to supply empty applications/manifestations where that's what it actually means?  If it's not required it would probably be better to drop it (and perhaps even turn these cases into explicit failure modes with nice exceptions).  I get a vague sense that `to_node` behaving this way may be a good way to hide some bugs.


(I tried running the test suite with a change to make this raise an exception instead of supply empty defaults but the test suite still doesn't get past `test_pull_image_if_necessary` for me :cry: ).